src/app/modules/onboarding/module.nim

```diff
@@ -233,6 +237,38 @@ method loginRequested*[T](self: Module[T], keyUid: string, loginFlow: int, dataJ
     error "Error finishing Login Flow", msg = e.msg
     self.view.accountLoginError(e.msg, wrongPassword = false)
 
+proc syncAppAndKeycardState[T](self: Module[T]) =
+  let kcEvent = self.view.getKeycardEvent()
+  if kcEvent.keycardInfo.keyUID == "":
+    return
+  let keypair = self.controller.getKeypairByKeyUidFromDb(kcEvent.keycardInfo.keyUID)
+  if keypair.isNil:
+    return
+  let cardMetadata = self.controller.getMetadata()
```

Comment:
Metadata is already available as part of keycard status:
https://github.com/keycard-tech/status-keycard-go/blob/efec05e104fa9af6c4e52a235a168771287c5d4e/internal/keycard_context_v2_state.go#L77

So you should be able to get it from `kcEvent`. Seems that it just wasn't parsed here in Nim.

Reply:
Ohh I haven't seen them in `KeycardEventDto` and then just added the call for that. I will keep the call in case we need to check metadata (like the need for check what's on keycard flow) and also attach metadata to `KeycardEventDto` type.